cbpfc.go

```diff
@@ -470,8 +471,9 @@ type packetGuardOpts struct {
 	// requiredGuard returns:
 	// - the packetGuard needed by insns
 	// - the number of instructions in insns covered by the guard.
-	//   The guard is assumed to be invalidated for the remaining / uncovered insns (eg RegX was clobbered for indirect guards).
+	//   The guard is assumed to be invalidated for the next insns (eg RegX was clobbered for indirect guards).
 	//   requiredGuard will be called until all instructions are covered.
+	//   0 for all instructions covered.
 	requiredGuard func(insns []instruction) (int, packetGuard)
```

Comment:
I think I would find a `bool` return value easier to understand. The `0` is kind of clever but imo just leads to subtle bugs.

Reply:
Good point. The triple return value didn't seem very clear so I added a new `requriedGuard` struct that lets us document what the different fields are.